src/Core/Environment.php

```diff
@@ -89,17 +89,20 @@ public static function increaseMemoryLimitTo($memoryLimit = -1)
             return true;
         }
 
-        // Check hard maximums
-        $max = static::getMemoryLimitMax();
-        if ($max > 0 && ($memoryLimit < 0 || $memoryLimit > $max)) {
-            $memoryLimit = $max;
-        }
 
         // Increase the memory limit if it's too low
         if ($memoryLimit < 0) {
             ini_set('memory_limit', '-1');
-        } elseif ($memoryLimit > $curLimit) {
-            ini_set('memory_limit', Convert::bytes2memstring($memoryLimit));
+        } else {
+            // Check hard maximums
+            $max = static::getMemoryLimitMax();
+            if ($max > 0 && ($memoryLimit < 0 || $memoryLimit > $max)) {
```

Comment:
`$memoryLimit < 0` is redundant here. You know that it's >= 0 from the prior condition.
```suggestion
            if ($max > 0 && $memoryLimit > $max) {
```

Reply:
I think the current implementation is correct as if `$memoryLimit < 0`, then it's greater than $max (as it's unlimited)

Reply:
haha, yeah, my bad, @ScopeyNZ is right, that seems to be a redundant check